internal/client/depsdevclient/deps_dev_client.go

```diff
@@ -335,7 +337,7 @@ func (d *DepsClient) getDependenciesForPurl(ctx context.Context, purl string) (*
 	packageInput, err := helpers.PurlToPkg(purl)
 	if err != nil {
 		logger.Infof("failed to parse purl to pkg: %s", purl)
-		return nil, fmt.Errorf("failed to parse purl to pkg: %w", err)
+		return nil, nil
```

Comment:
Can we add documentation on the nil case in the function comment and what the difference between nil and error is?

On a closer look...there doesn't seem to be a case where an error and nil defer. In this case would it make sense for the check to be within the user of this private function?

Reply:
done!